### PR TITLE
Update to current practice

### PR DESCRIPTION
**Pull request by pjhartzell**
**Related Issue(s):**
Issue #54 
Issue #17 

**Description:**
Quite a few edits for consistent use of pre-commit for formatting and linting. Remove docs and notebook.

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md). - N/A
- [ ] Examples are added to `examples/` - N/A

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Configuration file for pre-commit (https://pre-commit.com/).
+# Please run `pre-commit run --all-files` when adding or changing entries.
```
Comment (gadomski) [inline .pre-commit-config.yaml:2]: Can you run `pre-commit autoupdate` to ensure all these hooks are up-to-date w/ their latest versions?

**Review by gadomski — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        types_or: [jupyter, markdown, python, shell]
+  - repo: https://github.com/PyCQA/flake8
+    rev: 4.0.1
+    hooks:
+      - id: flake8
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v0.942
+    hooks:
+      - id: mypy
+        files: "^src/.*\\.py$|^tests/.*\\.py$|^scripts/.*\\.py$"
```
[the remark this answers is not in the document]
Comment (pjhartzell) [reply .pre-commit-config.yaml]: yep. done. works.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (pjhartzell) [reply .pre-commit-config.yaml:2]: yep. done.

File: scripts/test [hunk 3]
```diff
@@ -17,16 +17,15 @@ if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
...
         codespell -I .codespellignore -f \
             scripts/* \
             *.py ./**/*.py \
-            *.md \
-            docs/*.rst docs/**/*.rst \
-            docs/*.ipynb docs/**/*.ipynb
+            *.md
 
-        pytest --cov=stactools ${COVERAGE_DIRS}
+        pytest --cov=stactools tests
```
[the remark this answers is not in the document]
Comment (pjhartzell) [reply scripts/test]: ok, yeah, makes sense. limit the coverage to just the package being developed.

[the remark this answers is not in the document]
Comment (pjhartzell) [reply scripts/test]: updated.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        args:
+          - --namespace-packages
+          - --explicit-package-bases
+          - --ignore-missing-imports
```
[the remark this answers is not in the document]
Comment (pjhartzell) [reply .pre-commit-config.yaml]: Yep. I removed that. Related to this is where to put the mypy options: as args in the mypy hook in `pre-commit-config.yaml` or in `mypy.ini`. Current practice seems to put options in both places. I think `mypy.ini` is the way to go. It makes intuitive sense to look there for the options, and, to me, is easier to understand.

TLDR, the `--ignore-missing-imports` has been removed. Note that all options are now in `mypy.ini` as the `args` section in  `pre-commit-config.yaml` is eliminated.

**Review by gadomski — APPROVED** [review]

**Comment by pjhartzell** [comment]
Closes #54 
Closes #17
